examples/domainmodel/src/cli/generator.ts

```diff
@@ -0,0 +1,79 @@
+/******************************************************************************
+ * Copyright 2021 TypeFox GmbH
+ * This program and the accompanying materials are made available under the
+ * terms of the MIT License, which is available in the project root.
+ ******************************************************************************/
+
+import fs from 'fs';
+import _ from 'lodash';
+import { CompositeGeneratorNode, IndentNode, NL, processGeneratorNode } from 'langium';
+import { AbstractElement, Domainmodel, Entity, Feature, isEntity, isPackageDeclaration, Type } from '../language-server/generated/ast';
+
+export function generateJava(domainmodel: Domainmodel, fileName: string, destination: string = '.') {
+    const path = fileName.replace(/\..*$/, '').replaceAll(/[.-]/g, '');
+
+    generateAbstractElements(destination, domainmodel.elements, path);
+    return `${destination}/${path}`;
+}
+
+function generateAbstractElements(destination: string, elements: Array<AbstractElement | Type>, path: string): void {
+
+    function generateAbstractElementsInternal(elements: Array<AbstractElement | Type>, path: string) {
+        const fullPath = `${destination}/${path}`;
+        if (!fs.existsSync(fullPath)) {
+            fs.mkdirSync(fullPath, { recursive: true });
+        }
+
+        const packagePath = path.replaceAll('/', '.').replace(/^\.+/, '');
+        for (const elem of elements) {
+            if (isPackageDeclaration(elem)) {
+                generateAbstractElementsInternal(elem.elements, `${path}/${elem.name.replaceAll('.', '/')}`);
+            } else if (isEntity(elem)) {
+                const fileNode = new CompositeGeneratorNode();
+                fileNode.append(`package ${packagePath};`, NL, NL);
+                generateEntity(elem, fileNode);
+                fs.writeFileSync(`${fullPath}/${elem.name}.java`, processGeneratorNode(fileNode));
+            }
+        }
+    }
+
+    generateAbstractElementsInternal(elements, path);
+}
+
+function generateEntity(entity: Entity, fileNode: CompositeGeneratorNode): void {
+    const maybeExtends = entity.superType ? ` extends ${entity.superType.$refName}` : '';
+    fileNode.append(`class ${entity.name}${maybeExtends} {`, NL);
+    fileNode.indent(classBody => {
+        const featureData = entity.features.map(f => generateFeature(f, classBody));
+        featureData.forEach(([generateField, , ]) => generateField());
+        featureData.forEach(([, generateSetter, generateGetter]) => { generateSetter(); generateGetter(); } );
+    });
+    fileNode.append('}', NL);
+}
+
+function generateFeature(feature: Feature, classBody: IndentNode): [() => void, () => void, () => void] {
+    const name = feature.name;
```

Comment:
There is a problem when the feature name is the same as a Java keyword. For example, the qualified-names example can't be compiled because there is a feature named `this`:
```
example/qualifiednames/baz/E3.java:4: error: as of release 8, 'this' is allowed as the parameter name for the receiver type only
    private E4 this;
               ^
```
The solution could be a consistent renaming in such cases, e.g. using `_` as prefix (`_this`).

Reply:
I propose to leave generation how it is:
- If I use my DSL for generating Java classes, I know about Java keywords and can not use them as names
- It's necessary to consider all Java keywords
- If we support features renaming, we need to support it for classes too
- Maybe, renaming / adding prefix is not the most expected behaviour

However, names clashing is still a problem now. So, I renamed `this` --> `that`.